src/utils.js

```diff
@@ -33,17 +35,22 @@ const altWidgetMap = {
     updown: UpDownWidget,
     range: RangeWidget,
     hidden: HiddenWidget,
-  },
-  "date-time": {
-    hidden: HiddenWidget,
   }
 };
 
+const stringFormatWidgets = {
+  "date-time": DateTimeWidget,
+  "email": TextWidget, // XXX: to customize appropriately
+  "hostname": TextWidget,
+  "ipv4": TextWidget,
```

Comment:
I don't know, as I don't really want to write patterns for these (which can be tedious), and as maintaining the json schema should be fully done by the user of the lib.

Also, I think we're pretty much covered by our *jsonschema* lib dependency, which validates these natively.